src/rosinstall_generator/generator.py

```diff
@@ -250,10 +256,11 @@ def generate_rosinstall(distro_name, names,
         names.update(include_names_from_path)
 
     # Allow special keywords in repos
-    repo_names, keywords = _split_special_keywords(repo_names or [])
-    if set(keywords).difference(set([ARG_ALL_PACKAGES])):
+    repo_names, repo_keywords = _split_special_keywords(repo_names or [])
+    if set(repo_keywords).difference(set([ARG_ALL_PACKAGES])):
         raise RuntimeError('The only keyword supported by repos is %r' % (ARG_ALL_PACKAGES))
 
+    keywords = _merge_two_keywords(pkg_keywords, repo_keywords)
     if ARG_ALL_PACKAGES in keywords:
```

Comment:
I'm not sure that merging the keywords is the right thing yet, but regardless it looks like `keywords` is `None` after merging using `rosinstall_generator ALL --rosdistro melodic` in both python 2 and python 3.

```
$ rosinstall_generator ALL --rosdistro melodic
Traceback (most recent call last):
  File "/home/sloretz/env3/bin/rosinstall_generator", line 11, in <module>
    load_entry_point('rosinstall-generator', 'console_scripts', 'rosinstall_generator')()
  File "/home/sloretz/projects/rosinstall_generator/src/rosinstall_generator/cli.py", line 160, in main
    upstream_version_tag=args.upstream, upstream_source_version=args.upstream_development)
  File "/home/sloretz/projects/rosinstall_generator/src/rosinstall_generator/generator.py", line 264, in generate_rosinstall
    if ARG_ALL_PACKAGES in keywords:
TypeError: argument of type 'NoneType' is not iterable
```

```
$ rosinstall_generator ALL --rosdistro melodic
Traceback (most recent call last):
  File "/home/sloretz/env2/bin/rosinstall_generator", line 11, in <module>
    load_entry_point('rosinstall-generator', 'console_scripts', 'rosinstall_generator')()
  File "/home/sloretz/projects/rosinstall_generator/src/rosinstall_generator/cli.py", line 160, in main
    upstream_version_tag=args.upstream, upstream_source_version=args.upstream_development)
  File "/home/sloretz/projects/rosinstall_generator/src/rosinstall_generator/generator.py", line 264, in generate_rosinstall
    if ARG_ALL_PACKAGES in keywords:
TypeError: argument of type 'NoneType' is not iterable
```